openfl/experimental/workflow/runtime/federated_runtime.py

```diff
@@ -76,14 +114,154 @@ def collaborators(self) -> List[str]:
+        """
+        archive_path, exp_name = WorkspaceExport.export_federated(
+            notebook_path=self.notebook_path,
+            output_workspace="./generated_workspace",
+        )
+        return archive_path, exp_name
+
+    def submit_experiment(self, archive_path, exp_name) -> None:
+        """
+        Submits experiment archive to the director
+
+        Args:
+            archive_path (str): Archive file path containing the workspace.
+            exp_name (str): The name of the experiment to be submitted.
+        """
+        try:
+            response = self._dir_client.set_new_experiment(
+                archive_path=archive_path, experiment_name=exp_name, col_names=self.__collaborators
+            )
+            self.experiment_submitted = response.status
+
+            if self.experiment_submitted:
+                print(
+                    f"\033[92mExperiment {exp_name} was successfully "
+                    "submitted to the director!\033[0m"
+                )
+            else:
+                print(f"\033[91mFailed to submit experiment '{exp_name}' to the director.\033[0m")
+        finally:
+            self.remove_workspace_archive(archive_path)
+
+    def get_flow_state(self) -> Tuple[bool, Any]:
+        """
+        Retrieve the updated flow status and deserialized flow object.
+
+        Returns:
+            status (bool): The flow status.
+            flow_object: The deserialized flow object.
+        """
+        status, flspec_obj = self._dir_client.get_flow_state()
+
+        # Append generated workspace path to sys.path
+        # to allow unpickling of flspec_obj
+        sys.path.append(str(self.generated_workspace_path))
+        flow_object = dill.loads(flspec_obj)
+
+        return status, flow_object
+
+    def get_envoys(self) -> None:
+        """Prints the status of Envoys in a formatted way."""
+        # Fetch envoy data
+        envoys = self._dir_client.get_envoys()
+        DATETIME_FORMAT = "%Y-%m-%d %H:%M:%S"
+        now = datetime.now().strftime(DATETIME_FORMAT)
+
+        # Prepare the table headers
+        headers = ["Name", "Online", "Last Updated", "Experiment Running", "Experiment Name"]
+        # Prepare the table rows
+        rows = []
+        for envoy in envoys.envoy_infos:
+            rows.append(
+                [
+                    envoy.envoy_name,
+                    "Yes" if envoy.is_online else "No",
+                    datetime.fromtimestamp(envoy.last_updated.seconds).strftime(DATETIME_FORMAT),
+                    "Yes" if envoy.is_experiment_running else "No",
+                    envoy.experiment_name if envoy.experiment_name else "None",
+                ]
+            )
+        # Use tabulate to format the table
+        result = tabulate(rows, headers=headers, tablefmt="grid")
+        # Display the current timestamp
+        print(f"Status of Envoys connected to Federation at: {now}\n")
+        print(result)
```

Comment:
I suggest moving this function to the Director who can find and consolidate the status from the envoys. This way, the `FederatedRuntime` (and therefore the experiment owner) would only interact with the Director. WDYT?

Reply:
Clarification: Status of envoys that are connected to federation is maintained by Director only
This function only provides the functionality to query and get the list of envoys from Director. Intended use case is that user may want to confirm the number & identity (name) of participants before launching the experiment.